File: BB.Core/Model/ProductPicture.cs
using System;
using System.Collections.Generic;

namespace BB.Core.Model
{
    public class ProductPicture
    {
        public ProductPicture()
        {
            FacedProducts = new List<Product>();
            DefaultingCategory = new List<ProductCategory>();
        }

        public Guid ProductPictureId { get; set; }

        public string PictureUrl { get; set; }

        public int ProductId { get; set; }

        public virtual Product Product { get; set; }

        public virtual ICollection<Product> FacedProducts { get; set; }

        public virtual ICollection<ProductCategory> DefaultingCategory { get; set; }
    }
}

File: BB.Web/app/Model/Products/ProductPicture.ts
import p = require("../Products/Product");
import Product = p.Product;
import pc = require("../Products/ProductCategory");
import ProductCategory = pc.ProductCategory;

export class ProductPicture{
    public ProductPictureId: string;
    public PictureUrl: string;
    public ProductId?: number;
    public Product: Product;
    public FacedProduct: Product[];
    public DefaultingCategory: ProductCategory[];
}